drivers/storage/portworx/deployment.go

```diff
@@ -0,0 +1,670 @@
+		}
+		if t.cluster.Spec.CloudStorage.SystemMdDeviceSpec != nil &&
+			*t.cluster.Spec.CloudStorage.SystemMdDeviceSpec != "" {
+			args = append(args, "-metadata", *t.cluster.Spec.CloudStorage.SystemMdDeviceSpec)
+		}
+		if t.cluster.Spec.CloudStorage.MaxStorageNodes != nil {
+			args = append(args, "-max_drive_set_count",
+				strconv.Itoa(int(*t.cluster.Spec.CloudStorage.MaxStorageNodes)))
+		}
+		if t.cluster.Spec.CloudStorage.MaxStorageNodesPerZone != nil {
+			args = append(args, "-max_storage_nodes_per_zone",
+				strconv.Itoa(int(*t.cluster.Spec.CloudStorage.MaxStorageNodesPerZone)))
+		}
+	}
+
+	if t.cluster.Spec.SecretsProvider != nil &&
+		*t.cluster.Spec.SecretsProvider != "" {
+		args = append(args, "-secret_type", *t.cluster.Spec.SecretsProvider)
+	}
+
+	if t.startPort != defaultStartPort {
+		args = append(args, "-r", strconv.Itoa(int(t.startPort)))
+	}
+
+	if t.imagePullPolicy != v1.PullAlways {
+		args = append(args, "--pull", string(t.imagePullPolicy))
+	}
+
+	if t.cluster.Annotations[annotationCSIVersion] != "" {
+		args = append(args, "--csiversion", t.cluster.Annotations[annotationCSIVersion])
+	}
+
+	if t.isPxDev {
+		args = append(args, "--dev")
+	}
+
+	if t.isPKS {
+		args = append(args, "--keep-px-up")
+	}
+
+	if t.cluster.Annotations[annotationLogFile] != "" {
+		args = append(args, "--log", t.cluster.Annotations[annotationLogFile])
+	}
+
+	if t.cluster.Annotations[annotationMiscArgs] != "" {
+		parts, err := shlex.Split(t.cluster.Annotations[annotationMiscArgs])
+		if err == nil {
+			args = append(args, parts...)
+		} else {
+			logrus.Warnf("error parsing misc args: %v", err)
+		}
+	}
+
+	return args
+}
+
+func (t *template) getEnvList() []v1.EnvVar {
+	envList := []v1.EnvVar{
+		{
+			Name:  "AUTO_NODE_RECOVERY_TIMEOUT_IN_SECS",
+			Value: "1500",
+		},
+		{
+			Name:  "PX_TEMPLATE_VERSION",
+			Value: templateVersion,
+		},
+	}
+
+	if t.isPKS {
+		envList = append(envList, v1.EnvVar{
+			Name:  "PRE-EXEC",
+			Value: "if [ ! -x /bin/systemctl ]; then apt-get update; apt-get install -y systemd; fi",
+		})
+	}
+
+	if CSI.isEnabled(t.cluster.Spec.FeatureGates) {
+		envList = append(envList, v1.EnvVar{
+			Name:  "CSI_ENDPOINT",
+			Value: "unix://" + csiBasePath + "/csi.sock",
+		})
+	}
+
+	if t.cluster.Spec.ImagePullSecret != nil && *t.cluster.Spec.ImagePullSecret != "" {
+		envList = append(envList, v1.EnvVar{
+			Name: "REGISTRY_CONFIG",
+			ValueFrom: &v1.EnvVarSource{
+				SecretKeyRef: &v1.SecretKeySelector{
+					// TODO: for older versions of k8s it could be .dockercfg
+					Key: ".dockerconfigjson",
+					LocalObjectReference: v1.LocalObjectReference{
+						Name: *t.cluster.Spec.ImagePullSecret,
+					},
+				},
+			},
+		})
+	}
+
+	for _, env := range t.cluster.Spec.Env {
+		envCopy := env.DeepCopy()
+		envList = append(envList, *envCopy)
+	}
+
+	return envList
+}
+
+func (t *template) getVolumeMounts() []v1.VolumeMount {
+	// TODO: Imp: add etcd certs to the volume mounts
```

Comment:
What will be current workaround without etcd cert support?

Reply:
No workaround for etcd with auth. Going to do soon in the next couple of PRs.